### PR TITLE
refactor(core): doc property

### PR DESCRIPTION
**Pull request by EYHN**
doc property upgraded to use orm.

The visibility of the property are simplified to three types: `always show`, `always hide`, `hide when empty`, and the default is `always show`.

CleanShot 2024-10-14 at 15 34 52

Added a sidebar view to manage properties
CleanShot 2024-10-14 at 15 35 58

new property ui in workspace settings
CleanShot 2024-10-14 at 15 36 44

Property lists can be collapsed
CleanShot 2024-10-14 at 15 37 59

### CODE REVIEW

**Comment by EYHN** [comment]
* **#8465** <a href="https://app.graphite.dev/github/pr/toeverything/AFFiNE/8465?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `canary`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @EYHN and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: packages/common/infra/src/modules/doc/entities/property-list.ts [hunk 1]
```diff
@@ -13,15 +14,59 @@ export class DocPropertyList extends Entity {
     []
   );
 
+  sortedProperties$ = this.properties$.map(list =>
+    // default index key is '', so always before any others
+    list.sort((a, b) => ((a.index ?? '') > (b.index ?? '') ? 1 : -1))
```
Comment (pengx17) [inline packages/common/infra/src/modules/doc/entities/property-list.ts]: `sort` -> `toSorted` for immutablility

File: packages/frontend/core/src/components/affine/page-properties/types/text.tsx [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+  useRef,
+  useState,
+} from 'react';
+
+import * as styles from './text.css';
+import type { PropertyValueProps } from './types';
+
+export const TextValue = ({ value, onChange }: PropertyValueProps) => {
+  const [tempValue, setTempValue] = useState<string>(value);
+  const handleClick = useCallback((e: React.MouseEvent) => {
```
Comment (pengx17) [inline packages/frontend/core/src/components/affine/page-properties/types/text.tsx:16]: we have `useCatchEventCallback` for this purpose

File: packages/frontend/core/src/components/affine/page-properties/icons/icons-selector.tsx [hunk 3]
```diff
@@ -1,57 +1,40 @@
...
-  }, []);
   const t = useI18n();
   return (
     <Scrollable.Root>
       <div role="heading" className={styles.menuHeader}>
         {t['com.affine.page-properties.icons']()}
       </div>
       <Scrollable.Viewport className={styles.iconsContainerScrollable}>
-        <div className={styles.iconsContainer} ref={ref}>
+        <div className={styles.iconsContainer}>
```
Comment (pengx17) [inline packages/frontend/core/src/components/affine/page-properties/icons/icons-selector.tsx:28]: why remove active state styling?

**Review by pengx17 — APPROVED** [review]
LGTM

**Pull request merged**